gateway/session_manager.go

```diff
@@ -133,20 +167,18 @@ func (l *SessionLimiter) limitRedis(currentSession *user.SessionState, key strin
 func (l *SessionLimiter) limitDRL(currentSession *user.SessionState, key string, rateScope string,
 	apiLimit *user.APILimit, dryRun bool) bool {
 
-	// In-memory limiter
-	if l.bucketStore == nil {
-		l.bucketStore = memorycache.New()
-	}
-
 	bucketKey := key + ":" + rateScope + currentSession.LastUpdated
 	currRate := apiLimit.Rate
 	per := apiLimit.Per
 
+	tokenValue := uint(l.drlManager.CurrentTokenValue())
+
 	// DRL will always overflow with more servers on low rates
-	rate := uint(currRate * float64(l.Gw.DRLManager.RequestTokenValue))
-	if rate < uint(l.Gw.DRLManager.CurrentTokenValue()) {
-		rate = uint(l.Gw.DRLManager.CurrentTokenValue())
+	rate := uint(currRate * float64(l.drlManager.RequestTokenValue))
```

Comment:
this conflicts with imported package `github.com/TykTechnologies/tyk/internal/rate`, shall we rename?

Reply:
Renamed to `cost` to avoid shadowing the import